app/controllers/data_access_controller.rb

```diff
@@ -74,58 +74,70 @@ def retrieval
 
     if orders.size > 0
       order_ids = orders.map {|o| o['order_id']}
-      order_response = order_ids.compact.size > 0 ? echo_client.get_orders({id: order_ids}, token) : nil
-      if order_response && order_response.success?
-        echo_orders = order_response.body.map {|o| o['order']}.index_by {|o| o['id']}
-
-        orders.each do |order|
-          echo_order = echo_orders[order['order_id']]
-          if echo_order
-            order['order_status'] = echo_order['state']
-          else
-            # echo order_id doesn't exist yet
-            order['order_status'] ||= 'creating'
+      order_ids.each do |order_id|
+          order_response = order_id && order_id.compact.size > 0 ? echo_client.get_orders({id: order_id}, token) : nil
```

Comment:
`order_ids` is the array of orders per collection. so it could look like this

```
[collection_1_order_id, [collection_2_id_1, collection_2_id_2]]
```

I'm not sure if it will work sending all of them at once